packages/edit-site/src/components/secondary-sidebar/inserter-sidebar.js

```diff
@@ -45,6 +65,32 @@ export default function InserterSidebar() {
 		libraryRef.current.focusSearch();
 	}, [] );
 
+	const handleSelectTab = useCallback(
+		( tab ) => {
+			// Only enter the zoom-out mode when it's not in the focus mode.
+			if ( ! hasEditorCanvasContainer && ! isFocusMode ) {
+				// Enter the zoom-out mode when selecting the patterns tab and exit otherwise.
+				if ( tab === 'patterns' ) {
+					setEditorMode( 'zoom-out' );
+				} else if ( getEditorMode() === 'zoom-out' ) {
+					setEditorMode( 'edit' );
+				}
+			}
+		},
+		[ getEditorMode, setEditorMode, isFocusMode, hasEditorCanvasContainer ]
+	);
+
+	const previousLocationRef = useRef( location );
+	useEffect(
+		function closeInserterOnNavigate() {
```

Comment:
Ideally, we should find a better way and place to manage this when we have a better routing system.